### ISSUE TITLE
There should be a api route and service functions that returns the default program

### ISSUE THREAD

**Issue by daveytech** [OpenTree-Education/rhizone-lms#415]
## Expected behaviour

There should be a router for programs that returns the default program template and its activities associated with the curriculum. 

## Details and resources

The router should be named programsRouter.

The service file should be called programsService and the function should be findProgram. It should fetch the default program and return it with activities.

Note- Data types will need to be updated in the type file. (api.d.ts)

## Checklist

- [x] This issue is linked to the appropriate project.
- [x] This issue has appropriate labels.

**Comment by seidior** [comment]
Based on our conversation today, let's modify the expected deliverables to include:

- one default route in programsRouter.ts
- the following four functions:
   - listPrograms(containingDate?: Date)
   - findProgram(programId: number)
   - listCurriculumActivities(curriculumId?: number)
   - findActivity(activityId: number)
- the appropriate data types for the api in webapp's `api.d.ts` file
   - required: Activity
   - required: Program
   - optional: Curriculum

**Comment by seidior** [comment]
Here's the current status of this ticket:
- All functionality works as implemented, to great celebration and cheer earlier today.
- All *enabled* tests are passing, but
- Tests for `listProgramsWithActivities` had to be disabled because of concurrency issues with the database testing.
- Finally, the last few commits properly handle time zones, which requires modifying the two existing migration files (usually a bad idea) and will require pull request #426 to be merged into `main` and merged into this branch first.

**Comment by seidior** [comment]
Updates for this ticket:
- With the team's decision to split out the listProgramsWithActivities functionality out into a separate ticket, we're able to pass all our tests and have full code coverage for our tests.
- With `luxon` support having been merged into `main`, we can rely on its functionality to get a proper hold of all of the time zone weirdness that is necessary, including in the MySQL connection.

Upon reviewing the ticket, I realized there are a couple more tasks that need to be done:
- We need to restrict the columns that we're retrieving from each table to only what is necessary to deliver the functionality requested. Once that is done, the models file and test files will need to be adjusted.
- The `api/src/models.d.ts` and `webapp/src/types/api.d.ts` files will need to be synchronized so that the webapp knows what functionality will be received from the frontend and cast values appropriately.

Some gotchas:
- One migration file had to be modified... again. Sorry.
- All of my testing was done on my dev machine which is in the `America/Los_Angeles` time zone, so I'm not 100% confident that the same results will be achieved by running this in another timezone. I'd appreciate the feedback!
- None of what we talked about this morning with regard to parallel code style with the rest of the codebase has been addressed.
